### ISSUE TITLE
Add CSS Grid Layout Module Level 3 (Masonry!)

### ISSUE THREAD

**Issue by Schweinepriester** [Fyrd/caniuse#5677]
<https://drafts.csswg.org/css-grid-3/>

Probably early und likely will come via MDN, but keeping track 0:-)

**Comment by Schweinepriester** [comment]
<https://www.smashingmagazine.com/native-css-masonry-layout-css-grid/>

**Comment by CrazyStevenz** [comment]
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Masonry_Layout

**Comment by Schweinepriester** [comment]
Indeed available via MDN: <https://caniuse.com/?search=masonry>

Currently in two entries:
- <https://caniuse.com/mdn-css_properties_masonry>
- <https://caniuse.com/mdn-css_properties_masonry-auto-flow>

**Comment by GPHemsley** [comment]
Indeed, this is covered by MDN data (via mdn/browser-compat-data#7249), but it appears there may be more work to be done.

`grid-template-rows: masonry` is covered here:
https://caniuse.com/mdn-css_properties_grid-template-rows_masonry

`masonry-auto-flow` (as mentioned above) is covered here:
https://caniuse.com/mdn-css_properties_masonry-auto-flow

But `grid-template-columns: masonry` is missing, and mdn/browser-compat-data#9078 suggests that it may actually be (incorrectly) covered by this entry:
https://caniuse.com/mdn-css_properties_masonry

**Comment by Schweinepriester** [comment]
<https://webkit.org/blog/13839/release-notes-for-safari-technology-preview-163/>:

> ### Masonry Layout
>
> - Enabled Masonry layout by default ([25869](https://commits.webkit.org/258694@main)[4@main](https://commits.webkit.org/258694@main), [259308@main](https://commits.webkit.org/259308@main))
> - Added the initial support for `align-tracks` and `justify-tracks` ([258544@main](https://commits.webkit.org/258544@main))

**Comment by nathanchase** [comment]
https://github.com/w3c/csswg-drafts/pull/13191

> Update CSS-Grid-3 spec to use new grid-lanes syntax

<img width="2814" height="532" alt="Image" src="https://github.com/user-attachments/assets/ff2222f1-f40e-4421-87ec-b76b75806810" />

**Comment by jensimmons** [comment]
Yes! Now we need a page for Grid Lanes. The future is clear.

**Comment by karlhorky** [comment]
Thanks @jensimmons, I see you and others wrote a WebKit blog post about CSS Grid Lanes now:

- https://webkit.org/blog/17660/introducing-css-grid-lanes/

**Comment by karlhorky** [comment]
CSS-Tricks (@sunkanmii) compiled some (early) compatibility data also on 19 Dec 2025 - see the "So, what's next?" section of the article:

- https://css-tricks.com/masonry-layout-is-now-grid-lanes/#aa-so-whats-next

**Comment by jensimmons** [comment]
I just submitted a PR: https://github.com/Fyrd/caniuse/pull/7451

**Comment by Fyrd** [comment]
Now available at https://caniuse.com/css-grid-lanes
